### DISCUSSION TITLE
How to handle API request data across pages?

### DISCUSSION THREAD

**Discussion by alex-ragin** [vercel/next.js#19339] [Help] [answered]
Hey guys,

I'm just starting exploring Next.js and migrating my app from create-react-app to Next.js. So far it looks amazing!

But I run into a problem with fetching data from a remote endpoint. In the current create-react-app, I'm requesting data within App.tsx (useEffect hook) which is then getting saved to redux store. When you navigate through the app, the page components select data from this store, therefore no additional API calls needed. This saves me tons of money as I host everything on AWS and instead of having an API call per page I have only one call at the initial app render.

So, I am curious how can I achieve the same result with Next.js? Are there any possibilities to have a similar approach with SSG?
If not, what would you recommend me in this case?

Any help is much appreciated. Thanks!

**Comment by jamesmosier** [comment, marked as answer]
Hi @alex-ragin. I'd recommend checking out the [with-redux](https://github.com/vercel/next.js/tree/canary/examples/with-redux/) example. This covers SSG as well as SSR with a Redux provider in the `_app.js` so it is shared between pages.

**Reply by alex-ragin** [reply]
Hi @jamesmosier, thanks a lot for your advice! I wasn't sure about using redux within the Next.js app as it seemed to me as an anti-pattern. But I'll definitely give it a try. Thanks!
